### ISSUE TITLE
False positive of “CA1508: Avoid dead conditional code” linked to rethrow and object creation (e.g. Stream)

### ISSUE THREAD

**Issue by bravequickcleverfibreyarn** [dotnet/roslyn-analyzers#4941]
### Analyzer

**Diagnostic ID**: [CA1508: Avoid dead conditional code](https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1508)

### Analyzer source

**SDK**: [Built-in CA analyzers in .NET 5 SDK or later](https://docs.microsoft.com/dotnet/fundamentals/productivity/code-analysis)

**Version**:  SDK 5.0.104

_AND_

**NuGet Package**: [Microsoft.CodeAnalysis.NetAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.NetAnalyzers)

**Version**: 5.0.3 (Latest)

### Describe the bug

CA1508 badly reports with link to rethrow and object initialization/creation in try block.

### Steps To Reproduce

Compare

(False positives)
```
fileStream fs = null;
try
{
    fs = File.Create("");
}
catch
{
    throw;
}
finally
{
    fs?.Dispose();
}

MemoryStream ms = null;
try
{
    ms = new MemoryStream();
}
catch
{
    throw;
}
finally
{
    ms?.Dispose();
}
```
to

(No issues)
```
FileStream fs1 = null;
try
{
    fs1 = File.Create("");
}
catch
{
    //throw;
}
finally
{
    fs1?.Dispose();
}

FileStream fs2 = null;
try
{
    fs2 = File.OpenRead("");
}
catch
{
    throw;
}
finally
{
    fs2?.Dispose();
}
```

### Expected behavior

There are now false positives on rethrow.

### Actual behavior

If `try` seems to “create” object, `finally` expects it `is not null` when `catch` `throw`s.

**Comment by mavasani** [comment]
Verified this is now fixed with #6828. Fix should be part of the next .NET 8 preview SDK
